pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -280,6 +285,7 @@ func (og *operationGenerator) GenerateDetachVolumeFunc(
 		}
 		if err != nil {
 			// On failure, add volume back to ReportAsAttached list
+			// TODO: I think we're not in a great state here (could the volume actually be detached?)  Also removing it and relying on re-adding it seems risky.
```

Comment:
The error means that detach is not issued so it is safe to add back. Remove it from the list before actually issuing the operation is because the race between master and node's controller. Some details in #40887